src/main/java/com/aws/greengrass/mqttclient/AwsIotMqttClient.java

```diff
@@ -252,8 +252,7 @@ int getTimeout() {
      * @param sessionPresent whether the session persisted
      */
     private synchronized void resubscribe(boolean sessionPresent) {
-        // Don't bother spin up a thread if subscription is empty
-        if (!subscriptionTopics.isEmpty()) {
+        if (currentlyConnected.get() && !subscriptionTopics.isEmpty()) {
```

Comment:
Do we need to handle exceptions from `currentlyConnected.get()`?

Reply:
`AtomicBoolean.get` shouldn't throw right? Actually I removed this one because `resubscribe` is called immediately after `currentlyConnected.set(true)`. The later one in `resubscribeDroppedTopicsTask` should be enough.